hera_pspec/testing.py

```diff
@@ -42,103 +46,108 @@ def build_vanilla_uvpspec(
     uvp = uvpspec.UVPSpec()
 
     Ntimes = 10
-    Nfreqs = 50
-    Ndlys = Nfreqs if Ndlys is None else Ndlys
-    Nspws = 1
-    Nspwfreqs = 1 * Nfreqs
-    Nspwdlys = 1 * Ndlys
+    
+    uvp.Nfreqs = 50
+    uvp.Ndlys = uvp.Nfreqs if Ndlys is None else Ndlys
+    uvp.Nspws = 1
+    uvp.Nspwfreqs = uvp.Nspws * uvp.Nfreqs
+    uvp.Nspwdlys = uvp.Nspws * uvp.Ndlys
 
     # [((1, 2), (1, 2)), ((2, 3), (2, 3)), ((1, 3), (1, 3))]
     blpairs = [101102101102, 102103102103, 101103101103]
     bls = [101102, 102103, 101103]
-    Nbls = len(bls)
-    Nblpairs = len(blpairs)
-    Nblpairts = Nblpairs * Ntimes
+    uvp.Nbls = len(bls)
+    uvp.Nblpairs = len(blpairs)
+    #uvp.Nblpairts = uvp.Nblpairs * Ntimes
 
-    blpair_array = np.tile(blpairs, Ntimes)
-    bl_array = np.array(bls)
-    bl_vecs = np.array(
+    time_array = np.linspace(2458042.1, 2458042.2, Ntimes)
+    uvp.time_1_array = np.tile(time_array, uvp.Nblpairs)
+    if equal_time_arrays:
+        uvp.time_2_array = uvp.time_1_array.copy()
+    else:
+        # Make time 2 array in-between time 1 array.
+        uvp.time_2_array = uvp.time_1_array + (time_array[1] - time_array[0]) / 2.0
+        
+    uvp.blpair_array = np.tile(blpairs, Ntimes)
+    uvp.bl_array = np.array(bls)
+    uvp.bl_vecs = np.array(
         [
             [5.33391548e00, -1.35907816e01, -7.91624188e-09],
             [-8.67982998e00, 4.43554478e00, -1.08695203e01],
             [-3.34591450e00, -9.15523687e00, -1.08695203e01],
         ]
     )
-    time_array = np.linspace(2458042.1, 2458042.2, Ntimes)
     lst_array = JD2LST(time_array, longitude=21.4283)
-    time_array = np.repeat(time_array, Nblpairs)
-    time_1_array = time_array
-    time_2_array = time_array
-    Ntpairs = len(set([(t1, t2) for t1, t2 in zip(time_1_array, time_2_array)]))
-    Nbltpairs = len(set([(blp, t1, t2) for blp, t1, t2 in zip(blpair_array, time_1_array, time_2_array)]))
-    lst_array = np.repeat(lst_array, Nblpairs)
-    lst_1_array = lst_array
-    lst_2_array = lst_array
-    time_avg_array = time_array
-    lst_avg_array = lst_array
-    spw_freq_array = np.tile(np.arange(Nspws), Nfreqs)
-    spw_dly_array = np.tile(np.arange(Nspws), Ndlys)
-    spw_array = np.arange(Nspws)
-    freq_array = np.linspace(100e6, 105e6, Nfreqs, endpoint=False)
-    dly_array = utils.get_delays(freq_array, n_dlys=Ndlys)
-    polpair_array = np.array(
-        [
-            1515,
-        ]
-    )  # corresponds to ('xx','xx')
-    Npols = len(polpair_array)
-    vis_units = "unknown"
-    norm_units = "Hz str"
-    weighting = "identity"
-    channel_width = np.ones(Nfreqs) * np.median(np.diff(freq_array))
-    history = "example"
-    taper = "none"
-    norm = "I"
-    git_hash = "random"
-    scalar_array = np.ones((Nspws, Npols), float)
+    #time_array = np.repeat(time_array, Nblpairs)
+    uvp.Ntpairs = len(set([(t1, t2) for t1, t2 in zip(uvp.time_1_array, uvp.time_2_array)]))
```

Comment:
This can simply be `uvp.Ntpairs = uvp.time_1_array.size`

Reply:
Oh I’m just saying, if you test your code with random lists instead of the time arrays, the output is always the same whether you use your complicated line or just the `time_1_array.size`

Reply:
The repeated pairs get counted several times? Here if you have two arrays (a1,a2,a3) and (b1,b2,b3) your set is {(a1,b1), (a2,b2), (a3,b3)} because if the `zip` so the length is 3 - same as your input arrays. Or did I miss something?

Reply:
Yes but in most cases the two arrays are like

```
time_1_array = (a1, a2, a3, a1, a2, a3)
time_2_array = (b1, b2, b3, b1, b2, b3)
```

where the second repetition corresponds to a different baseline pair (but the same set of times)

and in this case the length of the set will be 3 but the length of `time_1_array` is 6.